docs/learn/documentation/versioned/aws/kinesis.md

```diff
@@ -0,0 +1,82 @@
+
+To configure Samza to consume from Kinesis streams:
+
+```
+# define a kinesis system factory with your identifier. eg: kinesis-system
+systems.kinesis-system.samza.factory=org.apache.samza.system.eventhub.KinesisSystemFactory
+
+# kinesis system consumer works with only AllSspToSingleTaskGrouperFactory
+job.systemstreampartition.grouper.factory=org.apache.samza.container.grouper.stream.AllSspToSingleTaskGrouperFactory
+
+# define your streams
+task.inputs=kinesis-system.input0
+
+# define required properties for your streams
+systems.kinesis-system.streams.input0.aws.region=YOUR-STREAM-REGION
+systems.kinesis-system.streams.input0.aws.accessKey=YOUR-ACCESS_KEY
+sensitive.systems.kinesis-system.streams.input0.aws.secretKey=YOUR-SECRET-KEY
+
+The tuple required to access the Kinesis data stream must be provided, namely the fields `YOUR-STREAM-REGION`, `YOUR-ACCESS-KEY`, `YOUR-SECRET-KEY`.
+```
+
+#### Advanced Configuration
+
+Samza also provides the following Kinesis system specific passthrough configs:
+* [AWS client configs](http://docs.aws.amazon.com/AWSJavaSDK/latest/javadoc/com/amazonaws/ClientConfiguration.html)
+* [AWS Kinesis Client Library configs](https://github.com/awslabs/amazon-kinesis-client/blob/master/src/main/java/com/amazonaws/services/kinesis/clientlibrary/lib/worker/KinesisClientLibConfiguration.java)
+
+```
+# configure AWS client passthrough configs.
+systems.kinesis-system.aws.clientConfig.CLIENT-CONFIG=YOUR-CLIENT-CONFIG-VALUE
+
+# set a proxy
+systems.kinesis-system.aws.clientConfig.ProxyHost=YOUR-PROXY-HOST
+systems.kinesis-system.aws.clientConfig.ProxyPort=YOUR-PROXY-PORT
+
+# configure AWS Kinesis Client Library passthrough configs.
+systems.kinesis-system.streams.input0.aws.kcl.KCL-CONFIG=YOUR-KCL-CONFIG-VALUE
+
+# to reset checkpoint and read from the oldest (TRIM_HORIZON) or the default upcoming offset (LATEST), change the table name and set the read position.
+systems.kinesis-system.streams.input0.aws.kcl.TableName=YOUR-APP-TABLE-NAME
+systems.kinesis-system.streams.input0.aws.kcl.InitialPositionInStream=YOUR-INIT-READ-POSITION
+```
+
+#### Limitations
+
+The following limitations apply for Samza jobs consuming from Kinesis streams using the Samza consumer:
+* No support for stateful processing: It could be done as a second stage job after repartitioning to kafka in the first stage.
+* No support for broadcast streams.
+* Kinesis streams cannot be configured as bootstrap streams.
+* SystemStreamPartitionGroupers other than AllSspToSingleTaskGrouper are not supported.
```

Comment:
Kinesis streams must be used with the AllSspToSingleTaskGrouper. No other grouper is supported. (should also add links wherever applicable)